python/ml4ir/applications/ranking/t_test.py

```diff
@@ -0,0 +1,90 @@
+from scipy import stats
+import numpy as np
+
+
+def perform_click_rank_dist_paired_t_test(mean, variance, n):
+    """
+    Compute the paired t-test statistic and its p-value given mean, standard deviation and sample count
+    Parameters
+    ----------
+    mean: float
+        The mean of the rank differences for the entire dataset
+    variance: float
+        The variance of the rank differences for the entire dataset
+    n: int
+        The number of samples in the entire dataset
+
+    Returns
+    -------
+    t_test_stat: float
+        The t-test statistic
+    pvalue: float
+        The p-value of the t-test statistic
+    """
+    t_test_stat = mean / (np.sqrt(variance/n))
+    df = n - 1
+    pvalue = 1 - stats.t.cdf(np.abs(t_test_stat), df=df)
+    return t_test_stat*2, pvalue*2  # multiplying by 2 for two sided t-test
+
+
+def compute_stats_from_stream(diff, count, mean, M2):
+    """
+    Compute the running mean, variance for a stream of data.
+    src: Welford's online algorithm: https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
+
+    Parameters
+    ----------
+    diff: List
+        A batch of differences in rankings
+    count: int
+        Aggregates the number of samples seen so far
+    mean: float
+        Accumulates the mean of the rank differences so far
+    M2: float
+        Aggregates the squared distance from the mean
+
+    Returns
+    -------
+    count: int
+        The updated aggregate sample count
+    mean: float
+        The updated mean of the rank differences so far
+    M2: float
+        The updated squared distance from the mean
+    """
+    for i in range(len(diff)):
+        count += 1
+        delta = diff[i] - mean
+        mean += delta / count
+        delta2 = diff[i] - mean
+        M2 += delta * delta2
+    return count, mean, M2
+
+
+def t_test_log_results(t_test_stat, pvalue, rank_distribution_t_test_pvalue_threshold, logger):
```

Comment:
I understand that this was an ask from @lastmansleeping, still having a separate logging function feels strange. Not a blocker.

Reply:
I just didn't want to overcrowd the `evaluate()` function. That's all.

Reply:
I agree there is almost no functionality here. :|

Reply:
@mohazahran Sorry, can we move it back? I assumed you would be able to move the entire t-test (all calls+logging) to a separate helper method. But I guess, that's not simple.

Reply:
I think the better way to do this is define a `run_ttest()` in `RelevanceModel` and then implement it in `RankingModel` with the code here and in the current ttest block in `RankingModel`.

Populate and return a ttest metrics dictionary from that above function.

In evaluate(), simply call that method and do a `metrics_dict.update(whatever dictionary was returned)`